### DISCUSSION TITLE
Using forwardRef in Tooltip with TypeScript results in an error

### DISCUSSION THREAD

**Discussion by elarens** [mantinedev/mantine#2028] [Q&A]
### What package has an issue

@mantine/core

### Describe the bug

I tried to add a label to a tooltip dynamically, which resulted in the following error:
_Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?_

<img width="424" alt="LabelErr" src="https://user-images.githubusercontent.com/31107130/183230125-5e5f2fa2-b363-4d1e-9c4b-0ef26034f96e.png">

To fix the error based on its description, I tried adding a forwardRef according to Mantine documentation, which resulted in another error:
Type 'ForwardedRef<unknown>' is not assignable to type 'LegacyRef<HTMLDivElement> | undefined'.

<img width="517" alt="RefErr" src="https://user-images.githubusercontent.com/31107130/183230039-9811a7b4-e306-49af-92a4-22da7c7b303d.png">

I have limited experience with refs and don't know how to fix the error.

I'm using Mantine 5.0.3. version, which is currently the latest

### What version of @mantine/hooks page do you have in package.json?

^5.0.3

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

No

**Comment by rtivital** [comment]
Use generic typescript params: `forwardRef<HTMLDivElement, YourComponentProps>((props, ref) => (<div ref={ref} />))`

**Reply by elarens** [reply]
I tried the suggested approach but still didn't succeed in passing a tooltip label dynamically...

<img width="465" alt="Code1" src="https://user-images.githubusercontent.com/31107130/183245497-076815c8-556d-408a-8b2b-b0d0655397c1.png">

<img width="611" alt="Code2" src="https://user-images.githubusercontent.com/31107130/183245502-a0b76554-8c3e-405b-8219-999b9a1d90ca.png">

Tooltip label accepts a ReactNode and forwardRef generates ForwardRefExoticComponent that is not assignable to ReactNode

**Reply by elarens** [reply]
P.S. Solved the issue by using Popover
